src/python/pants/engine/target.py

```diff
@@ -2740,6 +2773,13 @@ def to_unparsed_address_inputs(self) -> UnparsedAddressInputs:
         )
 
 
+# If this predicate returns false, then the target's field's deps will be ignored.
+# NB: This has to be defined after SpecialCasedDependencies. Implementations are defined above.
+ShouldTraverseDepsPredicate = Callable[
+    [Target, Union[Dependencies, SpecialCasedDependencies]], bool
```

Comment:
Can you use the `|` syntax here? IMO it's easier to grok.

Reply:
it's not in an annotation, so will likely not work?

Reply:
I tried that first. But we're using python3.9, and `|` was a new feature in 3.10. We get away with using it in type hints thanks to `from __future__ import annotations`, but this is not a type hint. So, sadly, no we need `Union` until we update to 3.10 or higher.

Reply:
I think you can fix this, and move the declaration up if you use https://docs.python.org/3/library/typing.html#typing.TypeAlias (from `typing_extensions`)

So:
```python
ShouldTraverseDepsPredicate: TypeAlias = "Callable[[Target, Dependencies|SpecialCasedDependencies], bool]"
```

Reply:
Yes. `TypeAlias` worked like a charm. :smile: